pkg/smi/traffic_split.go

```diff
@@ -0,0 +1,71 @@
+package smi
+
+import (
+	"context"
+	"fmt"
+
+	smiSplitClient "github.com/servicemeshinterface/smi-sdk-go/pkg/gen/client/split/clientset/versioned"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/openservicemesh/osm-health/pkg/common"
+	"github.com/openservicemesh/osm-health/pkg/kuberneteshelper"
+)
+
+// Verify interface compliance
+var _ common.Runnable = (*TrafficSplitCheck)(nil)
+
+// TrafficSplitCheck implements common.Runnable
+type TrafficSplitCheck struct {
+	client      kubernetes.Interface
+	pod         *v1.Pod
+	splitClient smiSplitClient.Interface
+}
+
+// IsInTrafficSplit checks whether a pod is affected by an SMI traffic split
+func IsInTrafficSplit(client kubernetes.Interface, pod *v1.Pod, smiSplitClient smiSplitClient.Interface) TrafficSplitCheck {
+	return TrafficSplitCheck{
+		client:      client,
+		pod:         pod,
+		splitClient: smiSplitClient,
+	}
+}
+
+// Info implements common.Runnable
+func (check TrafficSplitCheck) Info() string {
+	return fmt.Sprintf("Checking whether pod %s participates in a traffic split", check.pod.Name)
+}
+
+// Run implements common.Runnable
+func (check TrafficSplitCheck) Run() error {
+	services, err := kuberneteshelper.GetMatchingServices(check.client, check.pod.ObjectMeta.GetLabels(), check.pod.Namespace)
+	if err != nil {
+		return err
+	}
+	trafficSplits, err := check.splitClient.SplitV1alpha2().TrafficSplits(check.pod.Namespace).List(context.TODO(), metav1.ListOptions{})
```

Comment:
Can you add a TODO here saying that in the future, we will decide the split version to use based on information dynamically obtained from the cluster?